internal/generate/generate.go

```diff
@@ -23,68 +37,79 @@ const (
 )
 
 var (
-	locodeGenerateInPaths        []string
+	locodeGenerateInPaths        stringSliceFlag
 	locodeGenerateSubDivPath     string
 	locodeGenerateAirportsPath   string
 	locodeGenerateCountriesPath  string
 	locodeGenerateContinentsPath string
 	locodeGenerateOutPath        string
+)
 
-	locodeGenerateCmd = &cobra.Command{
-		Use:   "generate",
-		Short: "Generate UN/LOCODE database",
-		Args:  cobra.NoArgs,
-		Run: func(cmd *cobra.Command, _ []string) {
-			locodeDB := csvlocode.New(
-				csvlocode.Prm{
-					Path:       locodeGenerateInPaths[0],
-					SubDivPath: locodeGenerateSubDivPath,
-				},
-				csvlocode.WithExtraPaths(locodeGenerateInPaths[1:]...),
-			)
-
-			airportDB := airportsdb.New(airportsdb.Prm{
-				AirportsPath:  locodeGenerateAirportsPath,
-				CountriesPath: locodeGenerateCountriesPath,
-			})
-
-			continentsDB := continentsdb.New(continentsdb.Prm{
-				Path: locodeGenerateContinentsPath,
-			})
-
-			targetDB := locode.New(locodeGenerateOutPath)
-
-			names := &namesDB{
-				DB:    airportDB,
-				Table: locodeDB,
-			}
-
-			err := locode.FillDatabase(locodeDB, airportDB, continentsDB, names, targetDB)
-			if err != nil {
-				cmd.PrintErrln(err)
-			}
-		},
+func init() {
+	flag.Var(&locodeGenerateInPaths, locodeGenerateInputFlag, "List of paths to UN/LOCODE tables (CSV)")
```

Comment:
have you considered `flag.Func`? not necessary, just do not like additional structs (and code lines) when they may be omitted

Reply:
I agree with you, `flag.Func` is better because there are no complex handling and there is no need to create an additional type.